service/controller/v14patch2/resource/pod/delete.go

```diff
@@ -51,6 +51,15 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 			resourcecanceledcontext.SetCanceled(ctx)
 			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
 
+			return nil
+		}
+		areContainersTerminated := key.ArePodContainersTerminated(currentPod)
+		if areContainersTerminated {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "pod is already drained")
```

Comment:
This statement might not be always true, especially when we implement this purposefully as fallback in case draining does not work properly.